src/shared_strings.c

```diff
@@ -16,6 +16,12 @@
  * Forward declarations.
  */
 
+STATIC int _element_cmp(struct sst_element *element1,
+    struct sst_element *element2);
+
+LXW_RB_GENERATE_ELEMENT(sst_rb_tree, sst_element, sst_tree_pointers,
+    _element_cmp);
```

Comment:
You can’t as the first thing in the struct is the index, not the string.  The strcmp compares element->string.  strcmp also expects a char * and not an element *.

Martin

> On Dec 11, 2015, at 9:01 AM, John McNamara <notifications@github.com> wrote:
> 
> In src/shared_strings.c <https://github.com/jmcnamara/libxlsxwriter/pull/41#discussion_r47357083>:
> 
> > @@ -16,6 +16,12 @@
> >   * Forward declarations.
> >   */
> >  
> > +STATIC int _element_cmp(struct sst_element *element1,
> > +    struct sst_element *element2);
> > +
> > +LXW_RB_GENERATE_ELEMENT(sst_rb_tree, sst_element, sst_tree_pointers,
> > +    _element_cmp);
> We could probably just pass strcmp() here directly without a wrapper, right?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/jmcnamara/libxlsxwriter/pull/41/files#r47357083>.
>